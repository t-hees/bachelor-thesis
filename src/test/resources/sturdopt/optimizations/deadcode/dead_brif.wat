(module
  (func (export "main") (param i32) (result i32)
    (block (result i32)
      (block (result i32)
        i32.const 1
        local.get 0
        br_if 1 ;; This br_if jump target is alive
        drop
        i32.const 2
        i32.const 0
        br_if 1 ;; This br_if jump target is dead but it references a block with an alive label.
        ;; In the sturdy cfg the only following node is still an EndLabel (of the inner block)!
      )
      drop
      i32.const 3
    )
  )
)
