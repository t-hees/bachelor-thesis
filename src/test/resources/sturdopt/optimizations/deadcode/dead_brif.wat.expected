(module
  (func (export "main") (param i32) (result i32)
    (block (result i32)
      i32.const 1
      local.get 0
      br_if 0 ;; This br_if jump target is alive
      drop
      i32.const 2
      i32.const 0
      br_if 0 ;; Ideally this should be optimized to a drop
      drop
      i32.const 3
    )
  )
)
